### PR TITLE
Use StandardCharsets.UTF_8.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).

### CODE REVIEW

**Review by Stypox — DISMISSED** [review]
Mmmh, maybe utility methods `encodeUrlUTF8()` and `decodeUrlUTF8()` should be created in `Utils.java`, since I see `URLEncoder.encode(url, UTF_8)` being used quite often.

**Comment by Stypox** [comment]
***Reminder for myself***: open a PR that separates utility methods in more classes, once this is merged

**Comment by Isira-Seneviratne** [comment]
> [StandardCharsets.UTF_8 is](https://stackoverflow.com/questions/32102166/standardcharsets-utf-8-on-lower-api-lower-than-19)

It seems support for desugaring `StandardCharsets` is being added in a new version of the AGP: https://newpipe.net/blog/pinned/newpipe-legacy/

**Review by TobiGr — APPROVED** [review]
Changes look good to me.
Please rebase so we can see whether any new tests fail.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeTrendingExtractor.java [hunk 1]
```diff
@@ -35,14 +20,29 @@
...
+ * (at your option) any later version.
+ *
+ * NewPipe Extractor is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with NewPipe Extractor. If not, see <https://www.gnu.org/licenses/>.
+ */
```
Comment (FireMasterK) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeTrendingExtractor.java:45]: Did the linter move this here?

**Review by FireMasterK — APPROVED** [review]

**Review by TobiGr — APPROVED** [review]
I created a separate PR for the failing tests (#937). I'll merge this anyway

**Pull request merged**